core/ir/riscv64/codec.c

```diff
@@ -1815,9 +1816,7 @@ encode_u_immpc_opnd(instr_t *instr, byte *pc, int idx, uint32_t *out)
         imm = (byte *)opnd_get_instr(opnd)->offset - (byte *)instr->offset;
     else
         return false;
-    /* FIXME i#3544: Add an assertion here to ensure that the lower 12 bits of imm are all
-     * 0. Assert only if decode_info_t.check_reachable is true. We should mark it as false
-     * to skip the check in get_encoding_info(), as we did for AARCHXX. */
+    ASSERT(!di->check_reachable || ((imm >> 12) << 12) == imm);
```

Comment:
We should also add a test to verify that we fail gracefully for cases where check_reachable is true and the lower bits are not zero. Perhaps it makes more sense to return false for graceful failures (instead of the assert).

Reply:
Done. Thanks for the advice, I should've added more faulty tests.